src/io/flutter/view/FlutterView.java

```diff
@@ -430,24 +430,29 @@ private void presentDevTools(FlutterApp app, InspectorService inspectorService,
     final DevToolsManager devToolsManager = DevToolsManager.getInstance(app.getProject());
 
     if (devToolsManager.hasInstalledDevTools()) {
-      addBrowserInspectorViewContent(app, inspectorService, toolWindow, isEmbedded);
+      AsyncUtils.invokeLater(() -> addBrowserInspectorViewContent(app, inspectorService, toolWindow, isEmbedded));
     }
     else {
       presentLabel(toolWindow, INSTALLING_DEVTOOLS_LABEL);
-
       final CompletableFuture<Boolean> result = devToolsManager.installDevTools();
-      result.thenAccept(succeeded -> {
-        if (succeeded) {
-          addBrowserInspectorViewContent(app, inspectorService, toolWindow, isEmbedded);
-        }
-        else {
-          // TODO(helin24): Handle with alternative instructions if devtools fails.
-          presentLabel(toolWindow, DEVTOOLS_FAILED_LABEL);
-        }
-      });
+      awaitDevToolsInstall(app, inspectorService, toolWindow, isEmbedded, result);
     }
   }
 
+  protected void awaitDevToolsInstall(FlutterApp app, InspectorService inspectorService, ToolWindow toolWindow, boolean isEmbedded, CompletableFuture<Boolean> installResult) {
+    AsyncUtils.whenCompleteUiThread(installResult, (succeeded, throwable) -> {
+      if (succeeded) {
```

Comment:
watch out! succeeded is a Boolean so it can be null (when the throwable != null). 
That is why I use the pattern of checking whether the throwable is null first in these callbacks to avoid bugs.
You could also use something like
https://commons.apache.org/proper/commons-lang/javadocs/api-2.4/org/apache/commons/lang/BooleanUtils.html#isTrue(java.lang.Boolean)
or
if (succeeded != null && succeeded)